packages/camera/android/src/main/java/io/flutter/plugins/camera/CameraPlugin.java

```diff
@@ -469,19 +424,69 @@ private void prepareMediaRecorder(String outputFilePath) throws IOException {
       mediaRecorder = new MediaRecorder();
       mediaRecorder.setAudioSource(MediaRecorder.AudioSource.MIC);
       mediaRecorder.setVideoSource(MediaRecorder.VideoSource.SURFACE);
-      mediaRecorder.setOutputFormat(MediaRecorder.OutputFormat.MPEG_4);
-      mediaRecorder.setAudioEncoder(MediaRecorder.AudioEncoder.AAC);
-      mediaRecorder.setVideoEncoder(MediaRecorder.VideoEncoder.H264);
-      mediaRecorder.setVideoEncodingBitRate(1024 * 1000);
-      mediaRecorder.setAudioSamplingRate(16000);
-      mediaRecorder.setVideoFrameRate(27);
-      mediaRecorder.setVideoSize(videoSize.getWidth(), videoSize.getHeight());
+      mediaRecorder.setProfile(recordingProfile);
       mediaRecorder.setOutputFile(outputFilePath);
       mediaRecorder.setOrientationHint(getMediaOrientation());
 
       mediaRecorder.prepare();
     }
 
+    private CamcorderProfile getBestAvailableCamcorderProfileForResolutionPreset(
+        String resolutionPreset) {
+      int camcorderProfileIndex;
+      switch (resolutionPreset) {
+        case "veryHigh":
+          camcorderProfileIndex = 0;
+          break;
+        case "high":
+          camcorderProfileIndex = 1;
+          break;
+        case "medium":
+          camcorderProfileIndex = 2;
+          break;
+        case "low":
+          camcorderProfileIndex = 3;
+          break;
+        case "veryLow":
+          camcorderProfileIndex = 4;
+          break;
+        default:
+          throw new IllegalArgumentException("Unknown resolution preset: " + resolutionPreset);
+      }
+
+      int cameraId = Integer.parseInt(cameraName);
+      switch (camcorderProfileIndex) {
+        case 0:
```

Comment:
Wouldn't it be better if the user selected veryHigh but QUALITY_2160P isn't available on the device that it uses the next best solution? 
I think right now it would fall back to low quality.

Reply:
That's what is actually happening. This bit above is just to convert the preset from a string to an int. having anything else than one of the preset value here in an error. It should never happen since on the dart side creating the `CameraController` require a `ResolutionPreset` object, so it's impossible for users to make a mistake here. But people updating/improving the plugin could so it's cleaner to take it into account.

The bit under is actually trying to use the requested preset and reducing if the preset is not available on the device.

Reply:
Another precision, I did the same thing as what was already on the iOS side for the second switch (which is pretty smart BTW). The switch doesn't break or return unless the profile is available, which means it will try all the presets under the requested one if it's not available.